docs/02-app/01-building-your-application/02-data-fetching/03-server-actions-and-mutations.mdx

```diff
@@ -683,37 +685,77 @@ Remember to consider the [behavior and caveats](https://react.dev/reference/reac
 
 ### Error Handling
 
-When an error is thrown, it'll be caught by the nearest [`error.js`](/docs/app/building-your-application/routing/error-handling) or `<Suspense>` boundary on the client. We recommend using `try/catch` to return errors to be handled by your UI.
+#### User Errors
+
+Uncaught errors in Server Actions are considered to be internal server errors. As a result, in production, the error message is omitted and instead [a generic error with a digest](/docs/app/building-your-application/routing/error-handling#securing-sensitive-error-information) is sent to avoid leaking potentially sensitive information.
 
-For example, your Server Action might handle errors from creating a new item by returning a message:
+Therefore, if you want to return a user-friendly error message to be displayed in the UI, you should return a serializable object with the error message, rather than throwing an error:
 
 ```ts filename="app/actions.ts" switcher
 'use server'
 
-export async function createTodo(prevState: any, formData: FormData) {
-  try {
-    // Mutate data
-  } catch (e) {
-    throw new Error('Failed to create task')
+export async function signIn(formData: FormData) {
+  const passwordIsCorrect = await checkPassword(formData.get('password'))
+  if (!passwordIsCorrect) {
+    return { error: 'Incorrect password' }
```

Comment:
Nit: Return "Invalid Credentials", not just password for auth best practices. 

```suggestion
    return { error: 'Invalid email or password' }
```

Reply:
Good point! Thanks!